src/PanoImageRenderer/PanoImageRenderer.js

```diff
@@ -65,21 +66,31 @@ export default class PanoImageRenderer extends Component {
 		this.canvas = this._initCanvas(width, height);
 
 		this._image = null;
-		this._imageLoader = new ImageLoader();
 
-		this._onImageLoad = 	this._onImageLoad.bind(this);
-		this._onImageError = 	this._onImageError.bind(this);
+		this._onContentLoad = 	this._onContentLoad.bind(this);
+		this._onContentError = 	this._onContentError.bind(this);
 
-		this.setImage({image, imageType: sphericalConfig.imageType});
+		this.setImage({image, imageType: sphericalConfig.imageType, isVideo});
 	}
 
-	setImage({image, imageType}) {
+	setImage({image, imageType, isVideo = false}) {
+		this._isVideo = isVideo;
 		this._setImageType(imageType);
 
+		if (this._contentLoader) {
+			this._contentLoader.destroy();
+		}
+
+		if (isVideo) {
+			this._contentLoader = new VideoLoader();
+		} else {
+			this._contentLoader = new ImageLoader();
+		}
```

Comment:
contentLoader uses concept of inheritance. and use get/set common interface.

I've tried to combine imageLoader and videoLoader. But it is somewhat different logic. If being combined, code will be difficult to read.

Reply:
I agree with this.